src/SurfaceFluxes.jl

```diff
@@ -357,50 +363,83 @@ function obukhov_length(
     sc::AbstractSurfaceConditions{FT},
     uft::UF.AUFT,
     scheme;
-    tol::RS.AbstractTolerance = RS.SolutionTolerance(max(FT(1e-1), sqrt(eps(FT)))),
+    tol::RS.AbstractTolerance = RS.SolutionTolerance(FT(z_in(sc) / 50)),
     maxiter::Int = 10,
 ) where {FT}
-
-    function root_l_mo(x_lmo)
-        root = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
-        return root
-    end
-    sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), RS.CompactSolution(), tol, maxiter)
-    if sol.converged
-        L_MO = sol.root
-    else
-        thermo_params = SFP.thermodynamics_params(param_set)
-        if error_on_non_convergence()
-            KA.@print("-----------------------------------------\n")
-            KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
-            KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
-            KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
-            KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
-            KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
-            KA.@print(", u_in = ", u_in(sc))
-            KA.@print(", u_sfc = ", u_sfc(sc))
-            KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
-            KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
-            KA.@print(", Δz = ", Δz(sc))
-            error("Unconverged Surface Fluxes.")
-        else
-            # KA.@print("Warning: Unconverged Surface Fluxes. Returning neutral condition solution \n")
-            L_MO = sol.root
-        end
+    
+  function root_l_mo(x_lmo)
+      residual = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
+      return residual
+  end
+
+  thermo_params = SFP.thermodynamics_params(param_set)
+  grav = SFP.grav(param_set)
+  DSEᵥ_sfc = TD.virtual_dry_static_energy(thermo_params, ts_sfc(sc), grav * z_sfc(sc))
+  DSEᵥ_in = TD.virtual_dry_static_energy(thermo_params, ts_in(sc), grav * z_in(sc))
+  ΔDSEᵥ = DSEᵥ_in - DSEᵥ_sfc
+  ΔrDSEᵥ = ΔDSEᵥ / DSEᵥ_sfc
+
+  tol_neutral = FT(0.001)
+  function solve_lmo(ΔrDSEᵥ)
+    if abs(ΔrDSEᵥ) <= tol_neutral
+      L_MO = z_in(sc) / tol_neutral
+      sol = nothing
+      is_converged  = true
+    elseif ΔrDSEᵥ > tol_neutral
+      sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), RS.VerboseSolution(), tol, maxiter)
+      #sol = RS.find_zero(root_l_mo, RS.SecantMethod(z_in(sc)*tol_neutral, z_in(sc)/tol_neutral), 
```

Comment:
@tapios Here is Akshay's proposed change to SecantMethod. Do you have any suggestion?

Reply:
LGTM. But tol_neutral seems very small; 1e-2 would suffice I think.

Reply:
Thanks. Will change the threshold.

Reply:
@akshaysridhar Newton vs Secant: Does your current test suggest one is consistently better than the other? If so we can change to the better one. If not, I suggest that you choose one (as long as the convergence test passes), and we merge this first so that ClimaAtmos can use it, and experiment with it later. What do you think?

Reply:
Oh @tapios actually here it is relative tolerance for virtual DSE difference. Is a relative tolerance of 1e-2 still sufficient?

Reply:
Good point. No, we need lower tolerance for DSE. Maybe better for that to be absolute (e.g., 0.1 K x cp). And we should decouple this tolerance from the initial values on length for the secant iterations. You can use fixed values for those.

Reply:
Tolerance for DSE is now 0.1 K x cp.